src/prefect/engine/serializers.py

```diff
@@ -249,3 +261,95 @@ def _get_serialize_method(self, dataframe: "pd.DataFrame" = None) -> Callable:
             raise ValueError(
                 "Could not find serialization methods for {}".format(self.file_type)
             ) from exc
+
+
+class CompressedSerializer(Serializer):
+    """
+    A Serializer that wraps another Serializer and a compression function to serialize
+    Python objects with compression.
+
+    Args:
+        - serializer (Serializer): the serializer that this serializer wraps
+        - format (str): name of the selected pre-defined compression format (bz2, gzip,
+            lzma, or zlib)
+        - compress (Callable[..., bytes]): the custom compression function
+        - decompress (Callable[..., bytes]): the custom decompression function
+        - compress_kwargs (Dict[str, Any]): keyword arguments to be passed to the
+            compression function
+        - decompress_kwargs (Dict[str, Any]): keyword arguments to be passed to the
+            decompression function
+    """
+
+    def __init__(
+        self,
+        serializer: Serializer,
+        format: str = None,
+        compress: Callable[..., bytes] = None,
+        decompress: Callable[..., bytes] = None,
+        compress_kwargs: Dict[str, Any] = None,
+        decompress_kwargs: Dict[str, Any] = None,
+    ):
+        self._serializer = serializer
+
+        if format:
+            try:
+                self._compress, self._decompress = COMPRESSION_FORMATS[format]
+            except KeyError as e:
+                raise ValueError("unknown compression format: {}".format(format)) from e
+
+            if (compress and compress != self._compress) or (
+                decompress and decompress != self._decompress
+            ):
+                raise ValueError(
+                    "Both compression format and custom compression/decompression "
+                    "functions provided but are incompatible."
+                )
+        elif compress and decompress:
+            self._compress = compress
+            self._decompress = decompress
+        else:
+            raise ValueError(
+                "CompressedSerializer must be provided the compression format or the "
+                "compression/decompression functions."
```

Comment:
The traceback will already indicate that this is in `CompressedSerializer`. It's also clearer (IMO) to specify the kwarg names required rather than their meanings.

```suggestion
                "You must specify either `format` or `compress`/`decompress`"
```